test/models/policy_test.rb

```diff
@@ -207,6 +207,54 @@ class PolicyTest < ActiveSupport::TestCase
     end
   end
 
+  context '#clone_to' do
+    setup do
+      @canonical = profiles(:two)
+      @profile = Profile.new(parent_profile: @canonical,
+                             account: accounts(:one),
+                             policy: policies(:one)).fill_from_parent
+      policies(:one).update!(account: accounts(:one))
+      @profile.save!
+
+      @policy_host = PolicyHost.create!(
+        policy: @profile.policy,
+        host: hosts(:one)
+      )
+      @os_minor_version = '3'
+    end
+
+    should 'use existing profile' do
+      child_profile = @canonical.clone_to(
+        account: accounts(:one),
+        policy: @profile.policy,
+        os_minor_version: @os_minor_version
+      )
+      assert_equal @profile, child_profile
```

Comment:
Could we add assertions showing the `os_minor_version` does not get set on these profiles? Using `clone_to` and passing `os_minor_version` makes me nervous that it could set the os version during a report upload, but I don't think that could happen as written.

Since these three cases are for existing profiles, it may also make sense to wrap this with `assert_no_difference('Profile.count')`.

Reply:
Added.